app/controllers/katello/api/v2/content_view_versions_controller.rb

```diff
@@ -105,20 +106,33 @@ def export_api_status
       render json: { api_usable: SmartProxy.pulp_primary.pulp3_repository_type_support?(Katello::Repository::YUM_TYPE) }, status: :ok
     end
 
-    api :POST, "/content_view_versions/:id/export", N_("Export a content view version")
+    api :POST, "/content_view_versions/:id/export_legacy", N_("Export a content view version")
     param :id, :number, :desc => N_("Content view version identifier"), :required => true
-    param :destination_server, String, :desc => N_("Destination Server name, required for Pulp3")
+    param :destination_server, String, :desc => N_("Destination Server name, required for Pulp3"), :required => true
     param :chunk_size_mb, :number, :desc => N_("Chunk export-tarfile into pieces of chunk_size mega bytes. Relevant only for Pulp 3 repositories"), :required => false
     param :export_to_iso, :bool, :desc => N_("Export to ISO format. Relevant only for Pulp 2 repositories"), :required => false
     param :iso_mb_size, :number, :desc => N_("maximum size of each ISO in MB. Relevant only for Pulp 2 repositories"), :required => false
     param :since, Date, :desc => N_("Optional date of last export (ex: 2010-01-01T12:00:00Z). Relevant only for Pulp 2 repositories"), :required => false
+    def export_legacy
+      if SmartProxy.pulp_primary.pulp3_repository_type_support?(Katello::Repository::YUM_TYPE)
+        fail HttpErrors::BadRequest, _("Invalid usage for Pulp 2 repositories."\
+                                       "Use export-future for Yum repositories")
+      end
+      respond_for_async :resource => export_pulp_v2
+    end
+
+    api :POST, "/content_view_versions/:id/export", N_("Export a content view version")
+    param :id, :number, :desc => N_("Content view version identifier"), :required => true
+    param :destination_server, String, :desc => N_("Destination Server name, required for Pulp3"), :required => true
+    param :chunk_size_mb, :number, :desc => N_("Chunk export-tarfile into pieces of chunk_size mega bytes. Relevant only for Pulp 3 repositories"), :required => false
     def export
-      task = if SmartProxy.pulp_primary.pulp3_repository_type_support?(Katello::Repository::YUM_TYPE)
-               export_pulp_v3
-             else
-               export_pulp_v2
-             end
-      respond_for_async :resource => task
+      if SmartProxy.pulp_primary.pulp3_repository_type_support?(Katello::Repository::YUM_TYPE)
+        export_pulp_v3
+      else
+        export_pulp_v2
+      end
```

Comment:
This if block is not needed since the if condition next line should take care of it.

Reply:
Fixed